src/renderer/inputSystem/index.ts

```diff
@@ -70,107 +23,20 @@ const defaultActions: Action[] = [
-  //     controlService.send('TOGGLE')
-  //   },
-  // },
   {
-    name: 'flipperFront',
-    bindings: [
-      { type: 'gamepadBtnDown', button: buttonMappings.dpad.up },
-      // { type: 'keyboard', code: 'KeyI' },
-    ],
+    name: 'toggleArmControl',
+    bindings: [{ type: 'gamepadBtnDown', button: buttonMappings.start }],
     perform: () => {
-      flipperService.send('MODE_FRONT')
-    },
-  },
-  {
-    name: 'flipperBack',
-    bindings: [
-      { type: 'gamepadBtnDown', button: buttonMappings.dpad.down },
-      // { type: 'keyboard', code: 'KeyK' },
-    ],
-    perform: () => {
-      flipperService.send('MODE_REAR')
-    },
-  },
-  {
-    name: 'flipperRight',
-    bindings: [{ type: 'gamepadBtnDown', button: buttonMappings.dpad.right }],
-    perform: () => {
-      //TODO add check for arm control if in none flipper mode
-      flipperService.send('MODE_RIGHT')
-    },
-  },
-  {
-    name: 'flipperLeft',
-    bindings: [{ type: 'gamepadBtnDown', button: buttonMappings.dpad.left }],
-    perform: () => {
-      flipperService.send('MODE_LEFT')
-    },
-  },
-  {
-    name: 'switchForwardDirection',
-    bindings: [
-      { type: 'gamepadBtnDown', button: buttonMappings.back },
-      // { type: 'keyboard', code: 'KeyT', onKeyDown: true },
-    ],
-    perform: () => {
-      // TODO implement this client side by flipping the necessary axis direction see issue #82
-      // rosClient
-      //   .callService({ name: 'markhor/switch_direction' })
-      //   .catch(log.error)
-      store.dispatch(feedSlice.actions.switchDirection())
-      store.dispatch(inputSlice.actions.toggleReverse())
-    },
-  },
-  {
-    name: 'headlights',
-    bindings: [{ type: 'gamepadBtnDown', button: buttonMappings.Y }],
-    perform: () => {
-      rosClient.callService({ name: '/headlights' }).catch(log.error)
-    },
-  },
-  {
-    name: 'flipper_reset',
-    bindings: [{ type: 'gamepadBtnDown', button: buttonMappings.B }],
-    perform: () => {
-      rosClient
-        .callService({ name: 'markhor/flippers/flipper_reset' })
-        .catch(log.error)
-    },
-  },
-  {
-    name: 'spacemouse',
-    bindings: [{ type: 'spacemouse' }],
-    perform: (ctx) => {
-      if (ctx.type !== 'spacemouse') {
-        return
-      }
-      if (!controlService.state.matches('arm')) {
-        return
-      }
-
-      const joy = mapGamepadToJoy(ctx.gamepadState.gamepad)
-      rosClient.publish(spaceMouseTopic, joy)
-    },
-  },
-  {
-    name: 'gamepad',
-    bindings: [{ type: 'gamepad' }],
-    perform: (ctx) => {
-      if (ctx.type !== 'gamepad') {
-        return
+      controlService.send('TOGGLE')
+      if (controlService.state.matches('arm')) {
+        inputSystem.setActionMap(defaultActions.concat(armModeActions))
+      } else {
+        inputSystem.setActionMap(defaultActions.concat(flipperModeActions))
       }
-      const joy = mapGamepadToJoy(ctx.gamepadState.gamepad)
-      rosClient.publish(joyTopic, joy)
     },
   },
 ]
```

Comment:
Au lieu d'utiliser concat je ferais juste expand l'array directement, c'est un pattern que je vois plus souvent en js que d'utiliser concat, mais ca change pas grand chose, le résultat est le meme alors c'est comme tu veux.
```suggestion
...flipperModeActions
]
```

Reply:
I don't think using the spread operator is cleaner here. The spread operator is better than concat when adding single elements to existing arrays like `[5, ...numbers]`, but in our case we add 2 arrays so I don't think doing this is cleaner: [...defaultActions, ...flipperModeActions]